spring-apps/autosleep-core/src/main/java/org/cloudfoundry/autosleep/ui/web/controller/AutoEnrollmentController.java

```diff
@@ -0,0 +1,143 @@
+
+    @Autowired
+    private EnrollmentValidator validator;
+
+    @Autowired
+    private CloudFoundryApiService cloudfoundryApi;
+
+    @Autowired
+    private ParameterReader<Config.ServiceInstanceParameters.Enrollment> autoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Pattern> excludeFromAutoEnrollmentReader;
+
+    @Autowired
+    private ParameterReader<Duration> idleDurationReader;
+
+    @Autowired
+    private ParameterReader<EnrollmentConfig.EnrollmentParameters.EnrollmentState> stateReader;
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> getEnrolledOrganization(
+            @PathVariable String organizationId) {
+        log.debug("getEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<OrgEnrollmentConfig>(orgConfig, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> deleteOrganization(
+            @PathVariable String organizationId) {
+        log.debug("deleteEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        orgEnrollmentRepository.delete(organizationId);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> enrolOrganization(
+            @PathVariable String organizationId,
+            @RequestBody OrgEnrollmentConfigRequest orgEnrollerConfigRequest, BindingResult result)
+            throws CloudFoundryException, BindException {
+
+        log.debug("enrolOrganization - {}", organizationId);
+        if (!cloudfoundryApi.isValidOrganization(organizationId)) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        validator.validate(orgEnrollerConfigRequest, result);
+        if (result.hasErrors()) {
+            throw new BindException(result);
+        }
+
+        OrgEnrollmentConfig orgEnrollerConfig = OrgEnrollmentConfig.builder()
+                .organizationGuid(organizationId)
+                .idleDuration(orgEnrollerConfigRequest.getIdleDuration() == null ? null
+                        : idleDurationReader
+                                .readParameter(orgEnrollerConfigRequest.getIdleDuration(), false))
+                .excludeSpacesFromAutoEnrollment(
+                        orgEnrollerConfigRequest.getExcludeSpacesFromAutoEnrollment() == null ? null
+                                : excludeFromAutoEnrollmentReader
```

Comment:
> "@ valid" does not work for us if we want to accept the request object and validate it completely before returning.

It seems to me there is a fair amount of boilerplate code in this PR that would be removed/simplified by using the [spring validation support](https://docs.spring.io/spring/docs/current/spring-framework-reference/html/validation.html):
- EnrollmentValidator and its unit test
- Mapping from OrgEnrollmentConfigRequest to OrgEnrollmentConfig

I understand [spring mvc validation support](https://docs.spring.io/spring/docs/current/spring-framework-reference/html/mvc.html#mvc-config-validation
) makes it possible to return a list of all violations, and then to return it a details of the REST body response.

> Any validation violations will automatically be exposed as errors in the BindingResult accessible as a method argument and also renderable in Spring MVC HTML views.

/CC @s-bortolussi @o-orand